core/modules/filters/getvariablejson.js

```diff
@@ -0,0 +1,52 @@
+/*\
+title: $:/core/modules/filters/getvariablejson.js
+type: application/javascript
+module-type: filteroperator
+
+Filter operator to get widget variable info and
+Display as JSON with basic formatting
+
+\*/
+(function(){
+
+/*jslint node: true, browser: true */
+/*global $tw: false */
+"use strict";
+
+/*
+Export our filter function
+*/
+exports.getvariablejson = function(source,operator,options) {
+	var results = [],
+		widget = options.widget;
+	// "replacer" must be defined, otherwise JSON.stringify will throw a circular reference error RSOD
+	var replacer= "params name value default text isFunctionDefinition isProcedureDefinition isWidgetDefinition isMacroDefinition isVariable".split(" ");
+	source(function(tiddler,title) {
+		var text,
+			out = {},
+			info = widget.getVariableInfo(title, {});
+
+		out.text = info.text;
+		if (info.params) {
+			// unify parameter elements
+			out.params = info.params.map(function(param) {
```

Comment:
Please can you explain what this is doing and why it is necessary?

Reply:
The map function is needed, because there is an inconsistency in the core code. One of the variable definitions uses `param.value` and the others use `param.default` -- Or vice versa. I cannot remember without debugging info. I think we cannot change that behaviour without compatibility problems